src/main/java/org/candlepin/dto/shim/ProductInfoTranslator.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.shim;
+
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.ObjectTranslator;
+import org.candlepin.dto.api.v1.AttributeDTO;
+import org.candlepin.dto.api.v1.BrandingDTO;
+import org.candlepin.dto.api.v1.ContentDTO;
+import org.candlepin.dto.api.v1.ProductContentDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.service.model.BrandingInfo;
+import org.candlepin.service.model.ContentInfo;
+import org.candlepin.service.model.ProductContentInfo;
+import org.candlepin.service.model.ProductInfo;
+import org.candlepin.util.Util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The ProductDataTranslator provides translation from ProductData DTO objects to the new
+ * ProductDTOs (API)
+ */
+public class ProductInfoTranslator implements ObjectTranslator<ProductInfo, ProductDTO> {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ProductDTO translate(ProductInfo source) {
+        return this.translate(null, source);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ProductDTO translate(ModelTranslator translator, ProductInfo source) {
+        return source != null ? this.populate(translator, source, new ProductDTO()) : null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ProductDTO populate(ProductInfo source, ProductDTO destination) {
+        return this.populate(null, source, destination);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ProductDTO populate(
+        ModelTranslator modelTranslator, ProductInfo source, ProductDTO dest) {
+        if (source == null) {
+            throw new IllegalArgumentException("source is null");
+        }
+
+        if (dest == null) {
+            throw new IllegalArgumentException("dest is null");
+        }
+
+        dest.id(source.getId())
+            .uuid(source.getId())
```

Comment:
This should be null rather than being an incorrect value

Reply:
Done.